plugins/web/opentelemetry-instrumentation-user-interaction/src/instrumentation.ts

```diff
@@ -17,194 +17,218 @@
-    if (!element.getAttribute) {
-      return undefined;
-    }
-    if (element.hasAttribute('disabled')) {
-      return undefined;
-    }
-    if (!this._allowEventName(eventName)) {
-      return undefined;
-    }
-    const xpath = getElementXPath(element, true);
-    try {
-      const span = this.tracer.startSpan(
-        eventName,
-        {
-          attributes: {
-            [AttributeNames.COMPONENT]: this.component,
-            [AttributeNames.EVENT_TYPE]: eventName,
-            [AttributeNames.TARGET_ELEMENT]: element.tagName,
-            [AttributeNames.TARGET_XPATH]: xpath,
-            [AttributeNames.HTTP_URL]: window.location.href,
-            [AttributeNames.HTTP_USER_AGENT]: navigator.userAgent,
-          },
-        },
-        parentSpan
-          ? api.trace.setSpan(api.context.active(), parentSpan)
-          : undefined
-      );
+  private _patchAddEventListener() {
+    const plugin = this;
+    return (original: EventTarget['addEventListener']) => {
+      return function addEventListenerPatched(
+        this: HTMLElement,
+        type: keyof HTMLElementEventMap,
+        listener: EventListenerOrEventListenerObject | null,
+        options?: boolean | AddEventListenerOptions
+      ) {
+        if (
+          !listener ||
+          !plugin._allowEventName(type) ||
+          isTracingSuppressed(api.context.active())
+        ) {
+          return original.call(this, type, listener, options);
+        }
 
-      if (this._shouldPreventSpanCreation(eventName, element, span) === true) {
-        return undefined;
-      }
+        const element = this;
+        const patchedListener = function (event: Event) {
+          let span: api.Span;
+          let activeInteractionContext =
+            plugin._activeInteractionContexts.get(event);
+
+          if (!activeInteractionContext) {
+            // create active listener span
+            span = plugin.tracer.startSpan(type, {
+              startTime: event.timeStamp,
+              attributes: {
+                [AttributeNames.INSTRUMENTATION_NAME]: plugin.instrumentationName,
+                [AttributeNames.INSTRUMENTATION_VERSION]: plugin.instrumentationVersion,
+                [AttributeNames.EVENT_TYPE]: type,
+                [AttributeNames.TARGET_ELEMENT]: element.tagName,
+                [AttributeNames.TARGET_XPATH]: getElementXPath(element, true),
+                [AttributeNames.HTTP_URL]: window.location.href,
+                [AttributeNames.HTTP_USER_AGENT]: navigator.userAgent,
+              },
+            });
 
-      this._spansData.set(span, {
-        taskCount: 0,
-      });
+            if (plugin._zonePatched) {
+              // create a new zone for listener for given target
+              activeInteractionContext = Zone.current.fork(
+                new UserInteractionZoneSpec(
+                  span,
+                  function onComplete(attributes, events) {
+                    console.log('onComplete', event);
```

Comment:
When using this w/ react, I'm noticing that this onComplete function is executed right away and isn't being delayed until the async activity is completed. 

I updated your initial stack blitz with an [example here](https://stackblitz.com/edit/opentelemetry-js-interaction-with-gtm-iz9bho?file=src/app.tsx) if you'd like to take a look.